src/managers/KeybindManager.cpp

```diff
@@ -2013,6 +2014,163 @@ void CKeybindManager::pass(std::string regexp) {
+        return;
+    }
+
+    const std::string regexp = ARGS[2];
+    PHLWINDOW         PWINDOW    = nullptr;
+    const auto        LASTWINDOW = g_pCompositor->m_pLastFocus;
+
+    //if regexp is not empty, send shortcut to current window
+    //else, dont change focus
+    if (regexp != "") {
+        PWINDOW = g_pCompositor->getWindowByRegex(regexp);
+
+        if (!PWINDOW) {
+            Debug::log(ERR, "sendshortcut: window not found");
+            return;
+        }
+
+        if (!g_pSeatManager->keyboard) {
+            Debug::log(ERR, "No kb in sendshortcut?");
+            return;
+        }
+
+        if (!isMouse)
+            g_pSeatManager->setKeyboardFocus(PWINDOW->m_pWLSurface.wlr());
+        else
+            g_pSeatManager->setPointerFocus(PWINDOW->m_pWLSurface.wlr(), {1, 1});
+    }
+
+    //copied the rest from pass and modified it
+    g_pSeatManager->sendKeyboardMods(MOD, 0, 0, 0);
+
+    if (g_pKeybindManager->m_iPassPressed == 1) { //is the same variable as in pass, can be used, but without extra mouse button, handled above
+        if (!isMouse)
+            g_pSeatManager->sendKeyboardKey(g_pKeybindManager->m_uTimeLastMs, keycode - 8, WL_KEYBOARD_KEY_STATE_PRESSED);
+        else
+            g_pSeatManager->sendPointerButton(g_pKeybindManager->m_uTimeLastMs, keycode, WL_POINTER_BUTTON_STATE_PRESSED);
+    } else if (g_pKeybindManager->m_iPassPressed == 0) {
+        if (!isMouse)
+            g_pSeatManager->sendKeyboardKey(g_pKeybindManager->m_uTimeLastMs, keycode - 8, WL_KEYBOARD_KEY_STATE_RELEASED);
+        else
+            g_pSeatManager->sendPointerButton(g_pKeybindManager->m_uTimeLastMs, keycode, WL_POINTER_BUTTON_STATE_RELEASED);
+    } else {
+        // dynamic call of the dispatcher
+        if (!isMouse) {
+            g_pSeatManager->sendKeyboardKey(g_pKeybindManager->m_uTimeLastMs, keycode - 8, WL_KEYBOARD_KEY_STATE_PRESSED);
+            g_pSeatManager->sendKeyboardKey(g_pKeybindManager->m_uTimeLastMs, keycode - 8, WL_KEYBOARD_KEY_STATE_RELEASED);
+        } else {
+            g_pSeatManager->sendPointerButton(g_pKeybindManager->m_uTimeLastMs, keycode, WL_POINTER_BUTTON_STATE_PRESSED);
+            g_pSeatManager->sendPointerButton(g_pKeybindManager->m_uTimeLastMs, keycode, WL_POINTER_BUTTON_STATE_RELEASED);
+        }
+    }
+
+    if (PWINDOW) { //if focus of window was changed, this can't be still nullptr
+
+        //if focus was changed, get back
+
+        if (PWINDOW->m_bIsX11) { //xwayland hack, see pass
+            if (!isMouse) {
+                g_pSeatManager->state.keyboardFocus = nullptr;
+                g_pSeatManager->state.keyboardFocusResource.reset();
+            } else {
+                g_pSeatManager->state.pointerFocus = nullptr;
+                g_pSeatManager->state.pointerFocusResource.reset();
+            }
+        }
+
+        const auto SL = PWINDOW->m_vRealPosition.goal() - g_pInputManager->getMouseCoordsInternal();
+
+        if (!LASTWINDOW) {
```

Comment:
this if is wrong, remove it. If you found a window and focused to it, you need to get back to the last.

Reply:
Done